client/posts/execute-code-when-the-document-is-ready/index.tsx

```diff
@@ -7,7 +7,16 @@ export default () => {
 <Markdown
     content={`
 ~~~ javascript
-document.addEventListener('DOMContentLoaded', function(){
+const ready = function(cb) {
+    // Check if the \`document\` is loaded completely
+    (document.readyState === 'complete')
```

Comment:
I would probably change the check to the following:

```
(document.readyState === 'loading')
  ? document.addEventListener('DOMContentLoaded', cb)
  : cb();
```

IIRC this will cover the case when the DCL event is already fired but the `readyState` is not `complete` yet (e.g. it is `interactive`).

Reply:
One more remark. The `cb` is called inconsistently. When called directly it receives no parameters, but when it is passed as the event listener it receives the event object as a parameter.

This issue is minor though. Not sure what level of correctness is expected here :-)

Reply:
Thanks for the comments!
Regarding the level of correctness, of course, we always try our best.